test/jit/test_tracer.py

```diff
@@ -1407,6 +1406,47 @@ def forward(self, x):
 
         traced = torch.jit.trace(Wrapper(), (torch.rand(3, 4),))
 
+    def test_trace_multi_output_function(self):
+        # An autograd.Function with two outputs.
+        # It swaps inputs so we can check if shape
+        # handling is correct in TorchScript.
+        class Foo(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, x, y):
+                return y, x
+
+            @staticmethod
+            def backward(ctx, du, dv):
+                return dv, du
+
+        class Bar(torch.nn.Module):
+            def forward(self, x, y):
+                x = x.relu()
+                y = y.relu()
+                z = Foo.apply(x, y)
+                return z
+
+        x = torch.rand(3, 2)
+        y = torch.rand(1, 2)
+
+        traced = torch.jit.trace(Bar(), (x, y))
+
+        for n in traced.graph.nodes():
+            if n.kind() != 'prim::PythonOp':
+                continue
+            for o in n.outputs():
+                if o.type().kind() != 'TupleType':
+                    continue
+                for e in o.type().elements():
+                    # In this test, tensor elements in a tuple output 
+                    # can't have unknown shapes.
+                    assert e.dim()
```

Comment:
Nit:
Please use `self.assert***` methods instead of raw `assert`

Reply:
I am trying to replace those checks with `FileCheck`, so these `assert`s will be removed. Thanks.